src/Control/Parallel/Class.purs

```diff
@@ -19,13 +19,15 @@ import Data.Profunctor.Star (Star(..))
 import Effect.Class (class MonadEffect, liftEffect)
 import Effect.Ref as Ref
 
--- | The `Parallel` class abstracts over monads which support
--- | parallel composition via some related `Applicative`.
-class (Monad m, Applicative f) <= Parallel f m | m -> f, f -> m where
+-- | The `Parallel` class abstracts over pairs of `Apply`s where one of them
+-- | (`m`) composes sequentially, and the other (`f`) composes in parallel.
+-- | `m` is usually a `Monad`, which enforces the sequential nature of its
+-- | composition, but it doesn't need to be.
+class (Apply m, Apply f) <= Parallel f m | m -> f, f -> m where
   parallel :: m ~> f
   sequential :: f ~> m
 
-instance monadParExceptT :: Parallel f m => Parallel (Compose f (Either e)) (ExceptT e m) where
+instance monadParExceptT :: (Parallel f m, Apply (ExceptT e m)) => Parallel (Compose f (Either e)) (ExceptT e m) where
```

Comment:
These constraint changes don't seem right, `ExceptT e m` is already `Apply` if `m` is `Monad`: https://github.com/purescript/purescript-transformers/blob/be72ab52357d9a665cbf93d73ba1c07c4b0957ee/src/Control/Monad/Except/Trans.purs#L58

Same goes for `MaybeT`.

Reply:
But m isn't a Monad anymore, that's the point of the pr, and that's why i had to add these constraints

Reply:
Oh sorry, I misread the diff. In that case, the constraint here should be `Monad m`. It's impossible for there to be an `Apply (ExceptT e m)` instance otherwise anyway.

Reply:
Do you have an example of a case where this new formulation is useful, by the way?

Reply:
> the constraint here should be Monad m. It's impossible for there to be an Apply (ExceptT e m) instance otherwise anyway.

That's a good point, i didn't think of that. I thought `Monad m` would be more restrictive, but i see how in this function, it isn't. I agree that `Monad m` is clearer and better, in that case.

> Do you have an example of a case where this new formulation is useful, by the way?

My use case is that i want to try making a parser that is applicative, but not a monad, so that its structure can be observed without actually running it on something (and it'll be built up from a symbolic representation, kinda like explained towards the end of this post: https://cronokirby.com/posts/2020/10/lexical-combinators/).

Then i thought i could have a newtype around that, which instead composes parsers in a way that lets them parse in any order (for example, parsing command line arguments). This would be a good candidate for a `Parallel` instance, i thought. You'd be able to use `parApply` (or [`Par.ado`](https://pursuit.purescript.org/packages/purescript-qualified-do/2.2.0/docs/QualifiedDo.ParApply)) to make permutable sub-parsers in an overall sequential parser.

Basically i think the "parse these applicatively combined parsers in any order" is a good parallel analogy to "parse these applicatively combined parsers in the order they were combined", and there are reasons to make sequential parsers that are not monads.